plugins/optimization-detective/class-od-preload-link-collection.php

```diff
@@ -145,14 +166,7 @@ public function get_html(): string {
 		$link_tags = array();
 
 		foreach ( $this->get_adjacent_deduplicated_links() as $link ) {
-			$media_features = array( 'screen' );
-			if ( null !== $link['minimum_viewport_width'] && $link['minimum_viewport_width'] > 0 ) {
-				$media_features[] = sprintf( '(min-width: %dpx)', $link['minimum_viewport_width'] );
-			}
-			if ( null !== $link['maximum_viewport_width'] && PHP_INT_MAX !== $link['maximum_viewport_width'] ) {
-				$media_features[] = sprintf( '(max-width: %dpx)', $link['maximum_viewport_width'] );
-			}
-			$link['attributes']['media'] = implode( ' and ', $media_features );
+			$link = $this->add_media_features( (array) $link );
```

Comment:
Is the casting necessary?

```suggestion
			$link = $this->add_media_features( $link );
```

Reply:
Alternatively, if the above suggestion is accepted, then:

```suggestion
			$link['media] = $this->get_media_attribute( $link );
```